pkg/lib/authn/identity/loginid/provider.go

```diff
@@ -86,23 +86,13 @@ func (p *Provider) GetByValue(value string) ([]*identity.LoginID, error) {
 }
 
 func (p *Provider) GetByKeyAndValue(key string, value string) (*identity.LoginID, error) {
-	// Normalize expects loginID is in correct type so we have to validate it first.
-	invalid := p.Checker.ValidateOne(identity.LoginIDSpec{
-		Key:   key,
-		Type:  model.LoginIDKeyType(key),
-		Value: value,
-	}, CheckerOptions{
-		// Admin can create email login id which bypass domains blocklist allowlist
-		// it should not affect getting identity
-		// skip the checking when getting identity
-		EmailByPassBlocklistAllowlist: true,
-	})
+	cfg, ok := p.Config.GetKeyConfig(key)
 
-	if invalid != nil {
-		return nil, errors.Join(ErrValidate, invalid)
+	if !ok {
+		return nil, api.ErrGetUsersInvalidArgument.New("invalid Login ID key")
 	}
 
-	normalizer := p.NormalizerFactory.NormalizerWithLoginIDType(model.LoginIDKeyType(key))
+	normalizer := p.NormalizerFactory.NormalizerWithLoginIDType(cfg.Type)
 	normalizedloginID, err := normalizer.Normalize(value)
 	if err != nil {
 		return nil, errors.Join(ErrNormalize, err)
```

Comment:
You return `api.ErrGetUsersInvalidArgument.New("invalid Login ID value"` here. Because you know exactly the error is due to failed normalization.

Reply:
Removed `ErrNormalize` related code